From d7d2d4175939a0f5835f79479a6b0d0aae5487ec Mon Sep 17 00:00:00 2001
From: Quentin Casasnovas <quentinx.casasnovas@intel.com>
Date: Tue, 28 May 2013 11:17:53 +0200
Subject: [PORT FROM MAIN]wifi:SoftApController: Fix possible NULL dereference

BZ: 111389

Possible NULL deref.
Check if NULL before the deref.

Change-Id: I67a95c6aa5260f1e910eab5ebdc3fda0d37220ba
Orig-Change-Id: Ib62e4c9841eb081254bf94cd99af3ac4bdc11c35
Category: device enablement
Domain: CWS.WIFI-Common
Origin: internal
Upstream-Candidate: no,proprietary
Signed-off-by: Quentin Casasnovas <quentinx.casasnovas@intel.com>
---
 SoftapController.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/SoftapController.cpp b/SoftapController.cpp
index f87adec..f587503 100644
--- a/SoftapController.cpp
+++ b/SoftapController.cpp
@@ -384,6 +384,10 @@ int SoftapController::add_environment(const char *key, const char *val)
         if (!ENV_SOCKET[n]) {
             size_t len = strlen(key) + strlen(val) + 2;
             char *entry = (char *)malloc(len);
+
+            if (!entry)
+                return -ENOMEM;
+
             snprintf(entry, len, "%s=%s", key, val);
             ENV_SOCKET[n] = entry;
             return 0;
-- 
1.7.4.1

